2023/src/day5/almanac.rs

```diff
@@ -12,12 +14,88 @@ pub enum AmcEntryKind {
   HumidityToLocation,
 }
 
+pub struct Almanac {
+  pub seeds: Vec<u64>,
+  pub amc_entries: Vec<AlmanacEntry>,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct AlmanacPair {
```

Comment:
I actually almost did this and then realized `std::ops::Range` is the same thing lol, so I ended up using that.

`Range` is nice because you can call `contains()` on it, not sure if that would help you in `has_src_overlap`.

Reply:
oh it's identical lol - I'll keep this in mind for future. Thanks!